cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -385,9 +377,10 @@ writeLedgerState mOutFile qState@(Query.LedgerState serLedgerState) =
         $ LBS.writeFile fpath $ unSerialised serLedgerState
  where
    decodeLedgerState
-     :: Query.LedgerState era
-     -> Either LBS.ByteString (Ledger.NewEpochState ledgerera)
-   decodeLedgerState (Query.LedgerState (Serialised ls)) = first (const ls) (decodeFull ls)
+     :: Query.SerialisedLedgerState era
+     -> Either LBS.ByteString (Query.LedgerState era)
+   decodeLedgerState (Query.SerialisedLedgerState (Serialised ls)) =
+     first (const ls) (decodeFull ls)
```

Comment:
We should probably provide something like thisin the API itself to get from the `SerialisedLedgerState` to a `LedgerState` (or error), rather than having to know the underlying details of how it works here.

Reply:
I'll add this to my TODO list or I can delegate this as a small task.